code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeUtils.java

```diff
@@ -117,4 +122,81 @@ static AdobeError convertToAdobeError(final int errorCode) {
         }
         return error;
     }
+
+    /**
+     * Returns the mobile app surface
+     *
+     * @return {@link String} containing the mobile app surface
+     */
+    static String getMobileAppSurface() {
+        final String appSurface = ServiceProvider.getInstance().getDeviceInfoService().getApplicationPackageName();
+        if (OptimizeUtils.isNullOrEmpty(appSurface)) {
+            return null;
```

Comment:
We make a check after this to see if the prefixedSurface is a valid uri. If we return unknown, it will get prefixed as "mobileapp://unknown/surfaceName" which is a valid url and proceed to make a network call. Shouldn't be doing that right?

Reply:
It should be a rare occurrence, if at all. Also, server will not respond with any valid propositions in this case.

Reply:
Updated to return unknown